src/Quasar/QuasarF/Interpreter/Internal.purs

```diff
@@ -149,32 +148,48 @@ mkRequest
   . (String → Either Error a)
   → AXF.AffjaxF RequestContent String
   → Free (Coproduct l AXFP) (Either QError a)
-mkRequest f = map (handleResult f) <<< liftF <<< right
+mkRequest f a =
+  map (map snd) $ mkRequest' f a
+
+mkRequest'
+  ∷ ∀ a l
+  . (String → Either Error a)
+  → AXF.AffjaxF RequestContent String
+  → Free (Coproduct l AXFP) (Either QError (Tuple (AX.AffjaxResponse String) a))
+mkRequest' f = map (handleResult f) <<< liftF <<< right
+
+withExpired ∷ ∀ a b. Tuple (AX.AffjaxResponse a) b → ExpiredContent b
+withExpired (Tuple { headers: hs } content) =
+  ExpiredContent
+    { content
+    , expired: doSomethingWithHeaders hs
+    }
+  where
+  doSomethingWithHeaders headers = false
```

Comment:
@garyb what should I do with the headers? please help 🙏  😸

Reply:
Here `headers` has type `Array ResponseHeader` you can only [grab name and value ](https://pursuit.purescript.org/packages/purescript-affjax/5.0.0/docs/Network.HTTP.ResponseHeader#t:ResponseHeader) of `ResponceHeader`
So you would need to search for some header representing if a response is expired, than grab it's value, parse it and determine if it's expired. You might wanna take a look at this lines https://github.com/quasar-analytics/quasar/pull/2948/files#diff-04c6e90faac2675aa89e2176d2eec7d8L391

Reply:
yeah might still need help, I don't think I understand sorry 😢

Reply:
Which part you think you don't understand?